src/radio.c

```diff
@@ -0,0 +1,250 @@
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ * Authors:
+ *   rinigus <http://github.com/rinigus>
+
+ Radio switches are monitored, as exposed by rfkill interface in Linux
+ 
+ Sysfs description: https://www.kernel.org/doc/Documentation/ABI/stable/sysfs-class-rfkill
+ 
+ 
+**/
+
+#include "collectd.h"
+#include "common.h"
+#include "plugin.h"
+
+#define SYSFS_ROOT "/sys/class/rfkill/"
+#define BFSZ 512
+
+static int num_rfkill = 0;
+
+static int radio_init (void)
+{
+	int status;
+	char filename[256];
+
+	num_rfkill = 0;
+  
+	while (1)
+	{
+		status = ssnprintf (filename, sizeof (filename),
+							SYSFS_ROOT "rfkill%d/"
+							"hard", num_rfkill);
+      
+		if ((status < 1) || ((unsigned int)status >= sizeof (filename)))
+			break;
+      
+		if (access (filename, R_OK))
+			break;
+      
+		num_rfkill++;
+	}
+  
+	INFO ("radio plugin: Found %d radio%s", num_rfkill,
+		  (num_rfkill == 1) ? "" : "s");
+  
+	if (num_rfkill == 0)
+		plugin_unregister_read ("radio");
+  
+	return (0);
+}
+
+
+static void radio_submit (const char *type, const char *plugin_instance, const char *type_instance, gauge_t value)
+{
+	value_t values[1];
+	value_list_t vl = VALUE_LIST_INIT;
+  
+	values[0].gauge = value;
+  
+	vl.values = values;
+	vl.values_len = 1;
+	sstrncpy (vl.host, hostname_g, sizeof (vl.host));
+	sstrncpy (vl.plugin, "radio", sizeof (vl.plugin));
+	sstrncpy (vl.plugin_instance, plugin_instance, sizeof (vl.plugin_instance));
+	sstrncpy (vl.type, type, sizeof (vl.type));
+	sstrncpy (vl.type_instance, type_instance, sizeof (vl.type_instance));
+  
+	plugin_dispatch_values (&vl);
+}
+
+
+static _Bool get_value(const char *fname, int *value)
+{
+	FILE *fh;
+	char buffer[BFSZ];
+
+	if ((fh = fopen(fname, "r")) == NULL)
+		return (0);
+
+	if ( fgets(buffer, sizeof(buffer), fh) == NULL )
+    {
+		fclose(fh);
+		return (0); // empty file
+    }
+
+	(*value) = atoi( buffer );
+
+	fclose(fh);
+
+	return (1);
+}
+
+
+static _Bool get_string(const char *fname, char *buffer, size_t buffer_size)
+{
+	FILE *fh;
+	if ((fh = fopen(fname, "r")) == NULL)
+		return (0);
+
+	if ( fgets(buffer, buffer_size, fh) == NULL )
+    {
+		fclose(fh);
+		return (0); // empty file
+    }
+
+	fclose(fh);
+
+	return (1);
+}
+
+
+static void sanitize(char *buffer, size_t buffer_size)
+{
+	size_t i;
+	for (i=0; i<buffer_size; ++i)
+    {
+		char c = buffer[i];
+		if (c == '\0') return;
+		if (c == '\n')
+        {
+			buffer[i] = '\0';
+			return;
+        }
+      
+		if (i == buffer_size-1)
+        {
+			buffer[i] = '\0';
+			return;
+        }
+      
+		if (c == ' ' || c == '-' || c == '/')
+			buffer[i] = '_';
+    }
+}
+
+
+static int radio_read (void)
+{
+	char filename[BFSZ];
+	char type[BFSZ];
+	char name[BFSZ];
+	int status;
+	int hard;
+	int soft;
+	int i;
+
+	for (i=0; i < num_rfkill; ++i)
+    {
```

Comment:
indentation was off in several places. the second formatting commit should have addressed them all. at least emacs and less (from terminal) show decent result.